scrapy/crawler.py

```diff
@@ -35,8 +35,10 @@ def __init__(self, spidercls, settings):
 
         handler = LogCounterHandler(self, level=settings.get('LOG_LEVEL'))
         logging.root.addHandler(handler)
-        self.signals.connect(lambda: logging.root.removeHandler(handler),
-                             signals.engine_stopped)
+        # lambda is assigned to Crawler attribute because this way it is not
+        # garbage collected after leaving __init__ scope
+        self.__remove_handler = lambda: logging.root.removeHandler(handler)
```

Comment:
has to be an attribute? can't be just a variable ?
`__remove_handler = lambda: logging.root.removeHandler(handler)`